### PR TITLE
[#928] param store

### LINKED ISSUE #928: Consider common parameters store for CI\CD and tests
As a DevOps engineer I'd like to have a reliable unified storage for environment specific parameters so I don't have to store the same data across different parameters sources for CI\CD and tests.

### PR DESCRIPTION
**Pull request by alexander-semenets**
This fixes #928 issue.
Must be merged together with appropriate PRs in legion-infrastructure and legion-profiles repos.
**REMINDER:**
    GCP jobs must be updated according to temp _928 jobs after merge

### CODE REVIEW

**Review by maxkochubey — DISMISSED** [review]

**Review by mcdoker18 — APPROVED** [review]

File: containers/pipeline-agent/Dockerfile [hunk 1]
```diff
@@ -18,10 +18,11 @@ FROM python:3.6.8
 
 # Install python package dependencies and docker CLI
 RUN apt-get update && apt-get install -y software-properties-common \
-	&& apt-get install -y make build-essential libssl-dev libffi-dev zlib1g-dev libjpeg-dev git  \
+	&& apt-get install -y make build-essential libssl-dev libffi-dev zlib1g-dev libjpeg-dev git rubygems \
```
Comment (mcdoker18) [inline containers/pipeline-agent/Dockerfile]: Remove unused dependencies

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

File: legion/robot/legion/robot/profiler_loader.py [hunk 2]
```diff
@@ -14,92 +14,66 @@
...
-    if not profile:
-        profile = os.getenv(PROFILE_ENVIRON_KEY)
-        if not profile:
-            raise Exception('Cannot get profile - {} env variable is not set'.format(PROFILE_ENVIRON_KEY))
+    # load Cluster Profile
+    profile = profile or os.getenv(CLUSTER_PROFILE)
 
-    profile = os.path.abspath(os.path.join(arg, '{}.yml'.format(profile)))
+    if not profile:
+        raise Exception('Can\'t get profile at path {}'.format(profile))
```
Comment (kirillmakhonin) [inline legion/robot/legion/robot/profiler_loader.py:40]: It looks like it is old exception message. Message should be like "Cluster profile variable not set. Please set it or use env. variable {CLUSTER_PROFILE}"

File: legion/robot/legion/robot/profiler_loader.py [hunk 3]
```diff
@@ -14,92 +14,66 @@
...
+        except Exception as err:
+            raise Exception("Can\'t get variable from cluster profile: {}".format(err))
+
+        try:
+            init_session_id(variables['STATIC_USER_EMAIL'], variables['STATIC_USER_PASS'], host_base_domain)
+            variables['DEX_TOKEN'] = dex_client.get_token()
+            variables['DEX_COOKIES'] = dex_client.get_session_cookies()
+        except Exception as err:
+            raise Exception("Can\'t get dex authentication data: {}".format(err))
+        print(variables)
```
Comment (kirillmakhonin) [inline legion/robot/legion/robot/profiler_loader.py]: Is it secure to print all variables?

**Review by kirillmakhonin — CHANGES_REQUESTED** [review]

**Review by mcdoker18 — DISMISSED** [review]

File: legion/robot/legion/robot/profiler_loader.py (hunk 2 above)
Comment (alexander-semenets) [reply legion/robot/legion/robot/profiler_loader.py:40]: the idea of this message is to check if the profile argument has been really passed, the second try is to check if the profile presents on file system.

File: containers/pipeline-agent/Dockerfile (hunk 1 above)
Comment (alexander-semenets) [reply containers/pipeline-agent/Dockerfile]: done

File: legion/robot/legion/robot/profiler_loader.py (hunk 3 above)
Comment (alexander-semenets) [reply legion/robot/legion/robot/profiler_loader.py]: good catch, forgot about debug output

**Review by mcdoker18 — APPROVED** [review]

File: legion/robot/legion/robot/profiler_loader.py (hunk 2 above)
Comment (kirillmakhonin) [reply legion/robot/legion/robot/profiler_loader.py:40]: But as I think it's better to use `{!r}` notation in this case

**Review by kirillmakhonin — APPROVED** [review]
Pls see comments

**Pull request merged**